devlib/trace/ftrace.py

```diff
@@ -69,8 +75,17 @@ def __init__(self, target,
         self.start_time = None
         self.stop_time = None
         self.event_string = _build_trace_events(self.events)
+        self.function_string = _build_trace_functions(self.functions)
         self._reset_needed = True
 
+        # Setup tracing paths
+        self.available_functions_file = os.path.join(self.tracing_path, 'available_filter_functions')
```

Comment:
Should use `self.targe.path` rather than `os.path` for these as they are on-target paths and path handling on target and host may be different.